src/DataProvider.js

```diff
@@ -68,6 +68,23 @@ export default class DataProvider {
     }, this.polling())
   }
 
+  fetchWithRetry() {
```

Comment:
So, this really just retries to fetch the data only after the `getData` fails.

What if the request just hangs, and the response never comes? Or is that going to be handled in another PR?

Also, how does this play with `polling`, or calling the `refetch` manually?

Lastly, after this is finished, it really deserves a test.